app/scripts/views/mixins/avatar-mixin.js

```diff
@@ -197,6 +198,23 @@ define(function (require, exports, module) {
       this.notifier.triggerAll(Notifier.PROFILE_CHANGE, {
         uid: uid
       });
+    },
+
+    _supportsAvatarUpload () {
+      // There is a bug in iOS <= 10 where avatar upload fails when
+      // accessing from FxiOS settings webview.
+      // Ref: https://bugzilla.mozilla.org/show_bug.cgi?id=1334459
+      const userAgent = new UserAgent(this._getUserAgentString());
+      if (userAgent.isFirefoxIos()) {
+        const version = userAgent.parseOsVersion();
+        return version.major > 10;
+      } else {
+        return true;
+      }
+    },
+
+    _getUserAgentString () {
```

Comment:
To reduce the boilerplate here, you should be able to have the mixin declare a `dependsOn: [ UserAgentMixin ]` which will import all the right stuff for you. Then on [line 207](https://github.com/mozilla/fxa-content-server/pull/5480/files#diff-69c1b4c151f87a5c9ad048f78656cf9aR207) you can just call:

```js
const userAgent = this.getUserAgent();
```

For an example, see [the connect-another-device-mixin](https://github.com/mozilla/fxa-content-server/blob/462c0c8ab222c94040abef2dadf1d8e31d77ceb4/app/scripts/views/mixins/connect-another-device-mixin.js#L45).